### ISSUE TITLE
Discuss foreign programming language interop tools and libraries 

### ISSUE THREAD

**Issue by no-longer-on-githu-b** [LoungeCPP/loungesome-cpp#16]

**Comment by no-longer-on-githu-b** [comment]
- Boost.Python
- V8

**Comment by Morwenn** [comment]
There is a new C++11 alternative to Boost.Python that seems rather popular: [pybind11](https://github.com/wjakob/pybind11). As many things, I didn't try to use it but it might be of interest.

**Comment by nabijaczleweli** [comment]
@rightfold By "foreign language interop" you mean i18n or UTF-8 kinda stuff?

**Comment by milleniumbug** [comment]
No, look at the context. It obviously means programming language interop.

**Comment by nabijaczleweli** [comment]
Ah.

**Comment by ghost** [comment]
What are the states of lua binders? We need one for the list.

**Comment by no-longer-on-githu-b** [comment]
Well, Sol is still undocumented and full of special cases, so we'll have to find an awesome one instead.

**Comment by nabijaczleweli** [comment]
How about @ThePhD's [Sol2](https://github.com/ThePhD/sol2)?

**Comment by ThePhD** [comment]
@Morwenn is right introducing pybind11: it's way better than Boost.Python (and has the benefit of being specific to its domain and thus not encumbered with any of the usual old-old-old compiler workarounds usually demanded of boost libraries).

`Sol2` is going to undergo improvement and documentation over the course of the next 4 months. I'll post back here when sol v2.0.0 is "Loungesome Ready".

**Comment by milleniumbug** [comment]
Is it now?

**Comment by milleniumbug** [comment]
@ThePhD Is it now?

**Comment by ThePhD** [comment]
It's ready. :D https://www.reddit.com/r/cpp/comments/4a8gy7/sol2_lua_c_binding_framework/ https://github.com/ThePhD/sol2

**Comment by Morwenn** [comment]
On the other hand:

> [...] actually used AND appreciated in the industry [...]

**Comment by nabijaczleweli** [comment]
Somebody uses it (because it's good), and we (industry) appreciate it

**Comment by milleniumbug** [comment]
Well, since this list is a Lounge project, I thought this list was a list of projects with Lounge Seal of Quality.

Why? Well, it's mostly the loungers who vote there,  I can't speak for the entire industry.

**Comment by ThePhD** [comment]
Used in the industry before I even released:

> http://chat.stackoverflow.com/transcript/message/28817701#28817701

And there are many, many more individuals in the pool.

**Comment by Bulat-Ziganshin** [comment]
Sorry, but he said about Sol1. Or you just kidding? :)

**Comment by nabijaczleweli** [comment]
Lines between Sol and Sol2 are blurry, because one evolved from the other

**Comment by ThePhD** [comment]
Sol2 is meant to supersede Sol in every way, the only reason it's called Sol2 is because I had thought the original project maker abandoned the project and wanted nothing to do with it, so I went my own way.

After release day, he told me that he was going to give me push access to the original repo... but I had no way of knowing that, and with getting the impression they didn't want anything to do with it, I never asked. :/

**Comment by nabijaczleweli** [comment]
- [cereal](http://uscilab.github.io/cereal/)
